test/lib/ansible_test/_data/requirements/constraints.txt

```
@@ -2,6 +2,7 @@ coverage >= 4.5.1, < 5.0.0 ; python_version <  '3.7' # coverage 4.4 required for
 coverage >= 4.5.2, < 5.0.0 ; python_version == '3.7' # coverage 4.5.2 fixes bugs in support for python 3.7, coverage 5.0+ incompatible
 coverage >= 4.5.4, < 5.0.0 ; python_version >  '3.7' # coverage had a bug in < 4.5.4 that would cause unit tests to hang in Python 3.8, coverage 5.0+ incompatible
 cryptography < 2.2 ; python_version < '2.7' # cryptography 2.2 drops support for python 2.6
+cryptography < 3.2 ; python_version >= '2.7'  # cryptography 3.2 drops support for OpenSSL 1.0.2 which some of our CI hosts (FreeBSD) still use
```

how about being more precise?
```suggestion
cryptography < 3.2 ; python_version >= '2.7' and 'freebsd' in sys_platform  # cryptography 3.2 drops support for OpenSSL 1.0.2 which some of our CI hosts (FreeBSD) still use
```

---

Does that work (or at least not cause errors) back to pip 7.1?

---

Supported since 8.1.2: https://pip.pypa.io/en/stable/news/#id419

---

I discussed with mattclay, we would need it to work back to 7.1.